### PR TITLE
handle undefined entry as export specifier

### LINKED ISSUE #6551: 'export { undefined }' gives no emit, no error
```ts
export { undefined };
```

I'm not sure whether this should be an error or not, but apart from a `"use strict"` prologue, the output file is empty.

### PR DESCRIPTION
**Pull request by vladima**
fixes #6551

### CODE REVIEW

File: tests/cases/compiler/reExportUndefined2.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @module: commonjs
+
+// @filename: a.ts
+var undefined;
```
Comment (sandersn) [inline tests/cases/compiler/reExportUndefined2.ts]: did you mean to give the var the type `number`? It looks like you might be testing compatibility below when you say `use(undefined)`.

**Comment by sandersn** [comment]
:+1: once my question about the test is answered.

**Comment by vladima** [comment]
discussed offline

**Comment by mhegazy** [comment]
:+:

**Pull request merged**